rtgui/gradient.cc

```diff
@@ -399,7 +410,7 @@ bool Gradient::button1Pressed(int modifierKey)
 {
     EditDataProvider *provider = getEditProvider();
 
-    if (!(modifierKey & GDK_CONTROL_MASK)) {
+    if (!(modifierKey & GDK_CONTROL_MASK) && lastObject > -1) {
```

Comment:
If `lastObject > - 1` should always hold, I think it would be preferable to have
````
if (lastObject < 0) {
    return;
}
````
or even
````
assert(lastObject > -1);
````
at the beginning of the method.